main-actions/src/main/scala/sbt/Console.scala

```diff
@@ -41,7 +41,8 @@ final class Console(compiler: AnalyzingCompiler) {
       implicit log: Logger): Try[Unit] = {
     def console0() =
       compiler.console(classpath, options, initialCommands, cleanupCommands, log)(loader, bindings)
-    JLine.usingTerminal { _ =>
+    JLine.usingTerminal { t =>
+      t.init
```

Comment:
`usingTerminal` is (from :

```scala
  def usingTerminal[T](f: jline.Terminal => T): T =
    withTerminal { t =>
      t.restore
      val result = f(t)
      t.restore
      result
    }
```

(`withTerminal` ensures synchronised access to the terminal)

So this does `t.restore`, `t.init`, run console, `t.restore`.

Is that right?

Reply:
Yea. We need both `t.restore` *and* `t.init`, it seems.